File: PosBiptek/app/src/main/java/com/biptek/posbiptek/model/ProdukDiskon.java
package com.biptek.posbiptek.model;

public class ProdukDiskon {
    private long kode_diskon;
    private String kode_produk_diskon;
    private int besar_diskon;
    private String tanggal_mulai_berlaku_diskon;
    private String tanggal_berakhir_diskon;

    public ProdukDiskon(long kode_diskon, String kode_produk_diskon, int besar_diskon, String tanggal_mulai_berlaku_diskon, String tanggal_berakhir_diskon) {
        this.kode_diskon = kode_diskon;
        this.kode_produk_diskon = kode_produk_diskon;
        this.besar_diskon = besar_diskon;
        this.tanggal_mulai_berlaku_diskon = tanggal_mulai_berlaku_diskon;
        this.tanggal_berakhir_diskon = tanggal_berakhir_diskon;
    }

    public ProdukDiskon() {

    }

    public long getKode_diskon() {
        return kode_diskon;
    }

    public void setKode_diskon(long kode_diskon) {
        this.kode_diskon = kode_diskon;
    }

    public String getKode_produk_diskon() {
        return kode_produk_diskon;
    }

    public void setKode_produk_diskon(String kode_produk_diskon) {
        this.kode_produk_diskon = kode_produk_diskon;
    }

    public int getBesar_diskon() {
        return besar_diskon;
    }

    public void setBesar_diskon(int besar_diskon) {
        this.besar_diskon = besar_diskon;
    }

    public String getTanggal_mulai_berlaku_diskon() {
        return tanggal_mulai_berlaku_diskon;
    }

    public void setTanggal_mulai_berlaku_diskon(String tanggal_mulai_berlaku_diskon) {
        this.tanggal_mulai_berlaku_diskon = tanggal_mulai_berlaku_diskon;
    }

    public String getTanggal_berakhir_diskon() {
        return tanggal_berakhir_diskon;
    }

    public void setTanggal_berakhir_diskon(String tanggal_berakhir_diskon) {
        this.tanggal_berakhir_diskon = tanggal_berakhir_diskon;
    }

    @Override
    public String toString() {
        return "ProdukDiskon{" +
                "kode_diskon=" + kode_diskon +
                ", kode_produk_diskon='" + kode_produk_diskon + '\'' +
                ", besar_diskon=" + besar_diskon +
                ", tanggal_mulai_berlaku_diskon='" + tanggal_mulai_berlaku_diskon + '\'' +
                ", tanggal_berakhir_diskon='" + tanggal_berakhir_diskon + '\'' +
                '}';
    }
}

File: Server/advanced/backend/models/ProdukDiskon.php
<?php

namespace backend\models;

use Yii;

/**
 * This is the model class for table "produk_diskon".
 *
 * @property int $kode_diskon
 * @property string $kode_produk_diskon
 * @property int $besar_diskon(%)
 * @property string $tanggal_mulai_berlaku_diskon
 * @property string $tanggal_berakhir_diskon
 *
 * @property Produk $kodeProdukDiskon
 */
class ProdukDiskon extends \yii\db\ActiveRecord
{
    /**
     * {@inheritdoc}
     */
    public static function tableName()
    {
        return 'produk_diskon';
    }

    /**
     * {@inheritdoc}
     */
    public function rules()
    {
        return [
            [['kode_produk_diskon', 'besar_diskon(%)', 'tanggal_mulai_berlaku_diskon', 'tanggal_berakhir_diskon'], 'required'],
            [['besar_diskon(%)'], 'integer'],
            [['tanggal_mulai_berlaku_diskon', 'tanggal_berakhir_diskon'], 'safe'],
            [['kode_produk_diskon'], 'string', 'max' => 13],
            [['kode_produk_diskon'], 'exist', 'skipOnError' => true, 'targetClass' => Produk::className(), 'targetAttribute' => ['kode_produk_diskon' => 'kode_produk']],
        ];
    }

    /**
     * {@inheritdoc}
     */
    public function attributeLabels()
    {
        return [
            'kode_diskon' => 'Kode Diskon',
            'kode_produk_diskon' => 'Kode Produk Diskon',
            'besar_diskon(%)' => 'Besar Diskon(%)',
            'tanggal_mulai_berlaku_diskon' => 'Tanggal Mulai Berlaku Diskon',
            'tanggal_berakhir_diskon' => 'Tanggal Berakhir Diskon',
        ];
    }

    /**
     * @return \yii\db\ActiveQuery
     */
    public function getKodeProdukDiskon()
    {
        return $this->hasOne(Produk::className(), ['kode_produk' => 'kode_produk_diskon']);
    }
}
